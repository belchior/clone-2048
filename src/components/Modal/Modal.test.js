import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { Modal } from './index';

const initialProps = {
  bestScore: 0,
  button: { text: '', action: () => {}},
  hardMode: false,
  score: 0,
  title: '',
  wall: [],
};

const renderer = new ShallowRenderer();

it('Modal should render without crashing ', () => {
  const tree = renderer.render(<Modal {...initialProps} />);
  expect(tree).toMatchSnapshot();
});

it('Modal should render without crashing with all props', () => {
  const props = {
    bestScore: 2048,
    button: { text: 'Try Again', action: () => {}},
    hardMode: true,
    score: 234,
    title: 'You Won',
    wall: [2048, 0, 0, 0, 0, 0, 0, 0, 2, 4, 0, 0, 0, 0, 0, 0],
  };
  const tree = renderer.render(<Modal {...props} />);
  expect(tree).toMatchSnapshot();
});

it('button ModalAction when clicked should call buttonAction callback', () => {
  const buttonAction = jest.fn();
  const props = {
    ...initialProps,
    button: { text: 'Action', action: buttonAction },
  };
  const tree = renderer.render(<Modal {...props} />);
  const ModalAction = tree.props.children[1].props.children[3].props;

  ModalAction.onClick();
  expect(buttonAction.mock.calls).toHaveLength(1);
});
