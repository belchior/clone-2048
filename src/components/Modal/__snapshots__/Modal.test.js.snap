// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should render without crashing  1`] = `
<section
  className="Modal"
>
  <Wall
    hardMode={false}
    wall={Array []}
  />
  <div
    className="Modal-content"
  >
    <img
      alt="Logo Clone 2048"
      className="logo"
      src="clone-2048.svg"
    />
    <h1
      className="Modal-title"
    />
    <Score
      best={0}
      current={0}
    />
    <button
      className="ModalAction"
      onClick={[Function]}
      type="button"
    />
  </div>
</section>
`;

exports[`Modal should render without crashing with all props 1`] = `
<section
  className="Modal"
>
  <Wall
    hardMode={true}
    wall={
      Array [
        2048,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        2,
        4,
        0,
        0,
        0,
        0,
        0,
        0,
      ]
    }
  />
  <div
    className="Modal-content"
  >
    <img
      alt="Logo Clone 2048"
      className="logo"
      src="clone-2048.svg"
    />
    <h1
      className="Modal-title"
    >
      You Won
    </h1>
    <Score
      best={2048}
      current={234}
    />
    <button
      className="ModalAction"
      onClick={[Function]}
      type="button"
    >
      Try Again
    </button>
  </div>
</section>
`;
